### PR TITLE
Input process async

### PR DESCRIPTION
**Pull request by vincenthz**
* convert some tip/branches to async
* convert more input processing to async

### CODE REVIEW

File: jormungandr/src/blockchain/process.rs [hunk 1]
```diff
@@ -401,29 +298,91 @@ fn try_request_fragment_removal(
...
+    let storage = blockchain.storage().clone();
+
+    let tip_ref = tip.get_ref_std().await;
+
+    let tip_updated = if tip_ref.hash() == candidate.block_parent_hash() {
+        info!(
+            logger,
+            "update current branch tip: {} -> {}",
+            tip_ref.header().description(),
+            candidate.header().description(),
```
Comment (NicolasDP) [inline jormungandr/src/blockchain/process.rs]: you are repeating yourself with the logs here. 

create a log with the new context value instead. I.e:

```rust
let logger = logger.new(o!(
  "current_tip" => tip_ref.header().description(),
  "candidate" => tip.header().description(),
));
``` 

That will make the code shorter too and more easily readable.

File: jormungandr/src/blockchain/process.rs [hunk 2]
```diff
@@ -401,29 +298,91 @@ fn try_request_fragment_removal(
...
+    } else {
+        match chain_selection::compare_against(blockchain.storage(), &tip_ref, &candidate) {
+            ComparisonResult::PreferCurrent => {
+                info!(
+                    logger,
+                    "create new branch with tip {} | current-tip {}",
+                    candidate.header().description(),
+                    tip_ref.header().description(),
+                );
+                false
```
Comment (NicolasDP) [inline jormungandr/src/blockchain/process.rs]: no new branch is created here

Comment (vincenthz) [reply jormungandr/src/blockchain/process.rs]: indeed, but that's what aa7834627403e69984a4a1c4c55d33d945d40877 used to do

File: jormungandr/src/blockchain/process.rs (hunk 1 above)
Comment (vincenthz) [reply jormungandr/src/blockchain/process.rs]: just converting the old code, not doing any substantial improvement or anything different right now

**Pull request merged**